src/main/java/org/killbill/billing/plugin/analytics/dao/factory/BusinessAccountFactory.java

```diff
@@ -52,21 +52,28 @@ public BusinessAccountModelDao createBusinessAccount(final BusinessContextFactor
 
         // Retrieve the account balance
         // Note: since we retrieve the invoices below, we could compute it ourselves and avoid fetching the invoices
-        // twice, but that way the computation logic is owned by invoice
+        // twice, but that way the computation logic is owned by invoice (this will pull all invoices and items though as it's not optimized today)
```

Comment:
Since we already pull all the invoices, could we just compute the account balance ourselves, the same way KB is doing [it](https://github.com/killbill/killbill/blob/master/invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java#L591), as it is pretty much summing the invoice balances?

Reply:
Another thoughts, could we rely on analytics invoice tables to compute he account balance - as those (analytics invoices) get updated we could update the account balance?

Reply:
> Since we already pull all the invoices, could we just compute the account balance ourselves, the same way KB is doing [it](https://github.com/killbill/killbill/blob/master/invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java#L591), as it is pretty much summing the invoice balances?

As the comment hinted, I wanted to avoid this to keep the logic in Kill Bill (in case the logic becomes more complex over time).

> Another thoughts, could we rely on analytics invoice tables to compute he account balance - as those (analytics invoices) get updated we could update the account balance?

The main drawback with this approach is that Analytics refreshes are not _self-healing_ anymore, in case the state is corrupted for whatever reason.